### PR TITLE
Account overview page

### PR DESCRIPTION
**Pull request by bitbeckers**
* Load connected address and role
* Hooks for Hats metadata
* Updated `Header` with link to account page

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   },
   "dependencies": {
     "@hookform/resolvers": "^3.9.0",
+    "@radix-ui/react-avatar": "^1.1.1",
```
Comment (pheuberger) [inline package.json:17]: nit: I usually put adding a dependency into its own commit.

File: app/user/page.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+
+import { abi as HatsAbi } from "@/lib/hatsAbi";
+
+export default function AccountPage() {
+  const { address, isConnected } = useAccount();
+  const [hatId, setHatId] = useState<string | null>(null);
+
+  const hatsContractAddress = "0x3bc1A0Ad72417f2d411118085256fC53CBdDd137"; // Hats protocol contract address
```
Comment (pheuberger) [inline app/user/page.tsx:20]: Let's use a `constants.ts` file in the project root or something.

**Comment by pheuberger** [comment]
@bitbeckers the build is failing with

```
Failed to compile.
./app/user/page.tsx:31:12
Type error: Type 'string' is not assignable to type 'bigint'.
  29 |     abi: HatsAbi,
  30 |     functionName: "viewHat",
> 31 |     args: [hatsId],
     |            ^
  32 |     enabled: isConnected,
  33 |   });
  34 |
Error: Command "npm run build" exited with 1
```

File: app/user/page.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    enabled: isConnected,
+  });
+
+  useEffect(() => {
+    if (hatData) {
+      setHatId(hatData.toString());
+    }
+  }, [hatData]);
+
+  if (!isConnected) {
```
Comment (pheuberger) [inline app/user/page.tsx:40]: I'm thinking we should probably create a component out of this. We will need it in many places.

File: app/user/page.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    <div className="flex items-center justify-center min-h-screen p-4">
+      <Card className="w-full max-w-md">
+        <CardHeader>
+          <CardTitle>Account Details</CardTitle>
+          <CardDescription>Your Refunite network information</CardDescription>
+        </CardHeader>
+        <CardContent className="space-y-4">
+          <div className="flex items-center space-x-4">
+            <Avatar className="h-16 w-16">
+              <AvatarImage src={`https://robohash.org/${address}`} alt="User avatar" />
```
Comment (pheuberger) [inline app/user/page.tsx:65]: lol

File: app/user/page.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+          </div>
+
+          <div>
+            <h3 className="text-lg font-semibold mb-2">Hat Status</h3>
+            {isLoading ? (
+              <Skeleton className="h-8 w-24" />
+            ) : isError ? (
+              <Badge variant="destructive">Error loading Hat data</Badge>
+            ) : hatId ? (
+              <Badge variant="default">Hat ID: {hatId}</Badge>
```
Comment (pheuberger) [inline app/user/page.tsx:83]: The badge runs over by quite a bit 
image

File: app/user/page.tsx (hunk 2 above)
Comment (bitbeckers) [reply app/user/page.tsx:20]: Yes, but this is just components scaffolding

File: app/user/page.tsx (hunk 3 above)
Comment (bitbeckers) [reply app/user/page.tsx:40]: Let's do it when we need it

File: app/user/page.tsx (hunk 5 above)
Comment (bitbeckers) [reply app/user/page.tsx:83]: Not sure what I'm looking at, that data doesn't load for me yet as I didn't do much data fetching for the purpose of scaffolding

File: app/user/page.tsx (hunk 2 above)
Comment (pheuberger) [reply app/user/page.tsx:20]: Ah okay, no worries then :ok_hand:

File: app/user/page.tsx (hunk 5 above)
Comment (pheuberger) [reply app/user/page.tsx:83]: Alrighty :saluting_face:

**Comment by pheuberger** [comment]
Since Vercel doesn't work for some reason I checked out your PR and built it locally. Works fine. Let's squash the fix commit and shipit!

**Review by pheuberger — APPROVED** [review]
